### PR TITLE
DocumentDB guide

### PR DESCRIPTION
**Pull request by greedy52**
related:
- https://github.com/gravitational/teleport/pull/43867

### CODE REVIEW

File: docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+Log out of the database and remove credentials:
+
+```code
+$ tsh db logout rds-example
+```
+
+## Troubleshooting
+
+(!docs/pages/includes/database-access/aws-troubleshooting.mdx!)
```
Comment (GavinFrazar) [inline docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx:209]: This isn't the fault of this PR, but we should really split out or template the db specific troubleshooting.
I was surprised to see certificate error troubleshooting for RDS in the rendered preview.

Does DocumentDB have this issue? I assume there's some CA bundle for it, but I've never looked at DocumentDB before.

File: docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+`MONGODB-AWS` mechanisms for authentication.
+
+See [Authentication using IAM
+identity](https://docs.aws.amazon.com/documentdb/latest/developerguide/iam-identity-auth.html)
+for more details.
+
+## Next steps
+
+(!docs/pages/includes/database-access/guides-next-steps.mdx!)
+- Learn more about [Authentication using IAM identity for
```
Comment (GavinFrazar) [inline docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx:233]: generally, includes should always have whitespace before and after them.
It seems to be fine here, but sometimes it causes rendering issues.

```suggestion
(!docs/pages/includes/database-access/guides-next-steps.mdx!)

- Learn more about [Authentication using IAM identity for
```

**Review by GavinFrazar — APPROVED** [review]
This was very well done!

I'm not worried about the reference page changes - if you can get this to merge before my reference page revamp merges then I'll incorporate the DocumentdDB IAM permissions into the refactored discovery and database service IAM reference format.

File: docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+  </Admonition>
+- Command-line client `mongosh` or `mongo` installed and added to your
+  system's `PATH` environment variable.
+- A host, e.g., an EC2 instance, where you will run the Teleport Database
+  Service.
+- (!docs/pages/includes/tctl.mdx!)
+
+## Step 1/6. Create a Teleport user
+
+(!docs/pages/includes/database-access/create-user.mdx!)
```
[the remark this answers is not in the document]
Comment (greedy52) [reply docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx:44]: yes, it's at the protocol level.

> <Admonition type="warning">                                
>  Database names are only enforced for PostgreSQL, MongoDB, and Cloud Spanner databases.                                                   
> </Admonition> 

Are you thinking of updating this?

File: docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx (hunk 1 above)
Comment (greedy52) [reply docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx:209]: Good catch. DocumentDB uses the same bundle as RDS. I will update the error description there to include DocumentDB.

File: docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (GavinFrazar) [reply docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx:44]: yeah updating that to include docdb

[the remark this answers is not in the document]
Comment (greedy52) [reply docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx:44]: But we don't include like RDS there?

[the remark this answers is not in the document]
Comment (GavinFrazar) [reply docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx:44]: oh, I guess I keep forgetting that docdb is mongo - no need to update that then

File: docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+In this guide, we will use the example "teleport-docdb-user" for this role.
+
+Now click **Add new tag** at Step 3, use `TeleportDatabaseService` for the key
+and `Allowed` for the value. Then click **Create Role** to complete the process.
+
+![Create Role Step 3](../../../../img/database-access/guides/aws-create-role-add-tags.png)
+
+### Create a DocumentDB user
+
+Login your DocumentDB cluster with your master username and password from a
```
Comment (ptgott) [inline docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx:157]:
```suggestion
Log in to your DocumentDB cluster with your master username and password from a
```
Grammar suggestion

File: docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+machine that has network access to the DocumentDB cluster. Create a DocumentDB
+user with the IAM role ARN as username and specify `MONGODB-AWS` in the
+mechanisms for authentication:
```
Comment (ptgott) [inline docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx:160]:
```suggestion
mechanisms for authentication:

```

To make the source slightly easier to read

File: docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+$ tsh db logout rds-example
+```
+
+## Troubleshooting
+
+(!docs/pages/includes/database-access/aws-troubleshooting.mdx!)
+
+### Unable to authenticate using mechanism "MONGODB-AWS" error
+
+You may encounter the following error when IAM authentication fails:
```
Comment (ptgott) [inline docs/pages/enroll-resources/database-access/enroll-aws-databases/aws-docdb.mdx:213]:
```suggestion
You may encounter the following error when IAM authentication fails:

```

File: docs/pages/enroll-resources/database-access/reference/aws.mdx [hunk 7]
```diff
@@ -93,3 +103,4 @@ the related section below.
 (!docs/pages/includes/database-access/reference/aws-iam/redshift-serverless/role-as-user-policy.mdx dbUserRole="redshift-serverless-user-role" !)
 
 (!docs/pages/includes/database-access/iam_role_trust_relationship.mdx role1="teleport-db-service-role" role2="redshift-serverless-user-role"!)
+>>>>>>> 9cffa177ac3651968bb21907b6d46871ceb7889b
```
Comment (ptgott) [inline docs/pages/enroll-resources/database-access/reference/aws.mdx:106]: Stray conflict marker

File: docs/pages/includes/database-access/reference/aws-iam/documentdb/role-as-user-policy.mdx [hunk 8]
```diff
@@ -0,0 +1,6 @@
+The Teleport Database Service assumes a IAM role when connecting to a
+DocumentDB cluster with IAM authentication.
+
+Refer to the [Authentication using IAM
```
Comment (ptgott) [inline docs/pages/includes/database-access/reference/aws-iam/documentdb/role-as-user-policy.mdx:4]:
```suggestion
Refer to [Authentication using IAM
```

"The" doesn't refer to anything here. An alternative would be to keep "the" but say, "The Authentication using IAM Identity _guide_", so "the" would modify "guide".

**Review by ptgott — APPROVED** [review]
I've left mostly small grammar suggestions.

There's a lingering git conflict marker in `docs/pages/enroll-resources/database-access/reference/aws.mdx`.

**Pull request merged**